hnix-store-remote/src/System/Nix/Store/Remote/Client.hs

```diff
@@ -181,6 +182,18 @@ isValidPath
   -> m Bool
 isValidPath = doReq . IsValidPath
 
+-- | Download a NAR file.
+narFromPath
+  :: MonadRemoteStore m
+  => StorePath -- ^ Path to generate a NAR for
+  -> Word64 -- ^ Byte length of NAR
+  -> (ByteString -> IO()) -- ^ Data sink where NAR bytes will be written
+  -> m ()
+narFromPath path narSize sink = do
+  setDataSink sink
+  setDataSinkSize narSize
+  void $ doReq (NarFromPath path)
```

Comment:
If you always need to do `queryPathInfo` to figure out the size before, you might just add it here and possibly return its result as well.